dataclient/dataclient.go

```diff
@@ -0,0 +1,60 @@
+package dataclient
+
+import (
+	log "github.com/Sirupsen/logrus"
+	"github.com/zalando-incubator/skrop/filters"
+	"github.com/zalando/skipper/eskip"
+	"github.com/zalando/skipper/eskipfile"
+	"github.com/zalando/skipper/routing"
+)
+
+type skropDataClient struct {
+	fileName string
+	prepend  *eskip.Filter
+	append   *eskip.Filter
+}
+
+func NewSkropDataClient(eskipFile string) routing.DataClient {
+
+	emptyArgs := make([]interface{}, 0)
+
+	pre := &eskip.Filter{
+		Name: filters.NewFinalizeResponse().Name(),
+		Args: emptyArgs,
+	}
+	post := &eskip.Filter{
+		Name: filters.NewSetupResponse().Name(),
+		Args: emptyArgs,
+	}
+	return skropDataClient{
+		fileName: eskipFile,
+		prepend:  pre,
+		append:   post,
+	}
+}
+
+func (s skropDataClient) LoadAll() ([]*eskip.Route, error) {
+	f, err := eskipfile.Open(s.fileName)
+	if err != nil {
+		log.Error("Error while opening eskip file", err)
+		return nil, err
+	}
+
+	routes, err := f.LoadAll()
+	if err != nil {
+		log.Error("Error while loading eskip routes", err)
+		return nil, err
+	}
+
+	for _, route := range routes {
+		if route.BackendType != eskip.ShuntBackend {
```

Comment:
I have 2 concerns: the first is that the result would not change: even by checking if they are imagefilters, I would exclude some of the filters.
The second is that in the dataclient, the filters in the routes, are not skipper filters, but eskip.Filter, and they only contain a string which s the name of the filter. The only way for me to check if they are imagefilters is to compare all the strings of all the filters with that. isn't it ugly?